operator/builtin/input/aws/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,372 @@
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/observiq/stanza/errors"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/helper"
+)
+
+const operatorName = "aws_cloudwatch_input"
+const eventLimit = 10000 //The maximum number of events to return. The default is up to 10,000 events or max of 1mb.
+
+func init() {
+	operator.Register(operatorName, func() operator.Builder { return NewCloudwatchConfig("") })
+}
+
+// NewCloudwatchConfig creates a new AWS Cloudwatch Logs input config with default values
+func NewCloudwatchConfig(operatorID string) *CloudwatchInputConfig {
+	return &CloudwatchInputConfig{
+		InputConfig: helper.NewInputConfig(operatorID, operatorName),
+
+		EventLimit:   eventLimit,
+		PollInterval: helper.Duration{Duration: time.Minute * 1},
+		StartAt:      "end",
+	}
+}
+
+// CloudwatchInputConfig is the configuration of a AWS Cloudwatch Logs input operator.
+type CloudwatchInputConfig struct {
+	helper.InputConfig `yaml:",inline"`
+
+	// required
+	LogGroupName string `json:"log_group_name,omitempty" yaml:"log_group_name,omitempty"`
+	Region       string `json:"region,omitempty" yaml:"region,omitempty"`
+
+	// optional
+	LogStreamNamePrefix string          `json:"log_stream_name_prefix,omitempty" yaml:"log_stream_name_prefix,omitempty"`
+	LogStreamNames      []*string       `json:"log_stream_names,omitempty" yaml:"log_stream_names,omitempty"`
+	Profile             string          `json:"profile,omitempty" yaml:"profile,omitempty"`
+	EventLimit          int64           `json:"event_limit,omitempty" yaml:"event_limit,omitempty"`
+	PollInterval        helper.Duration `json:"poll_interval,omitempty" yaml:"poll_interval,omitempty"`
+	StartAt             string          `json:"start_at,omitempty" yaml:"start_at,omitempty"`
+}
+
+// Build will build a AWS Cloudwatch Logs input operator.
+func (c *CloudwatchInputConfig) Build(buildContext operator.BuildContext) ([]operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(buildContext)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.LogGroupName == "" {
+		return nil, fmt.Errorf("missing required %s parameter 'log_group_name'", operatorName)
+	}
+
+	if len(c.LogStreamNames) > 0 && c.LogStreamNamePrefix != "" {
+		return nil, fmt.Errorf("invalid configuration. Cannot use both 'log_stream_names' and 'log_stream_name_prefix' %s parameters", operatorName)
+	}
+
+	if c.Region == "" {
+		return nil, fmt.Errorf("missing required %s parameter 'region'", operatorName)
+	}
+
+	if c.EventLimit < 1 || c.EventLimit > 10000 {
+		return nil, fmt.Errorf("invalid value '%d' for %s parameter 'event_limit'. Parameter 'event_limit' must be a value between 1 - 10000", c.EventLimit, operatorName)
+	}
+
+	if c.PollInterval.Raw() < time.Second*10 {
+		return nil, fmt.Errorf("invalid value '%s' for %s parameter 'poll_interval'. Parameter 'poll_interval' must be a value of 10 seconds or greater", c.PollInterval.String(), operatorName)
+	}
```

Comment:
Is there a reason for 10 seconds?

Reply:
There is not a reason beyond 0 spams the AWS api to much, and I chose 10 seconds as the threshold. I am open to change for sure if it makes sense.

Reply:
This would be a good question for @djaglowski. I think we have avoided arbitrary limits in the past.

I agree it would likely be detrimental if the users set the limit too low (API access being rate limited).

We might need to re-work how the internal works. We spoke about this earlier, where maybe we have a separate go routine that is strictly polling the AWS api and handling the poll interval dynamically (with backoff, and only polling when the operator is ready to receive more messages). 

This PR is pretty big, I would support adding enhanced polling logic in a future PR.

Reply:
A lot of AWS APIs incur cost when being polled, so 10s may be completely reasonable. That said, unless we have a firm understanding of the cost dynamics, or a technical limitation, I think we should err on the side of allowing users more freedom. It is never fun for anyone when a software update is required instead of just a config change.

I suggest we use `1s` as the bare minimum.

Reply:
Sounds good I will make the change.